torch/_inductor/codegen/cpp.py

```diff
@@ -2532,9 +2550,28 @@ def reduction(self, dtype, src_dtype, reduction_type, value):
         self.reduction_prefix.writeline(
             f"{acc_type_vec} {acc_vec} = {self.reduction_init_vec(reduction_type, dtype)};"
         )
-        self.stores.writeline(
-            f"{acc_vec} = {self.reduction_combine_vec(reduction_type, acc_vec, value)};"
+        # save the reciprocal of weights for welford reduce if using static shape
```

Comment:
Can this optimization be applied to the dynamic shape too? The number of pre-computed weights can be decided at runtime.

Reply:
Yes. This optimization can be applied to the dynamic shape. Supported